{{cookiecutter.project_slug}}/config/settings/base.py

```diff
@@ -44,16 +44,28 @@
 # DATABASES
 # ------------------------------------------------------------------------------
 # https://docs.djangoproject.com/en/dev/ref/settings/#databases
-{% if cookiecutter.use_docker == "y" -%}
-DATABASES = {"default": env.db("DATABASE_URL")}
-{%- else %}
-DATABASES = {
-    "default": env.db(
-        "DATABASE_URL",
-        default="postgres://{% if cookiecutter.windows == 'y' %}localhost{% endif %}/{{cookiecutter.project_slug}}",
-    ),
-}
-{%- endif %}
+try:
+    {%- if cookiecutter.use_docker == "y" %}
+    DATABASES = {"default": env.db("DATABASE_URL")}
+    {%- else %}
+    DATABASES = {
+        "default": env.db(
+            "DATABASE_URL",
+            default="postgres://{% if cookiecutter.windows == 'y' %}localhost{% endif %}/{{cookiecutter.project_slug}}",
+        ),
+    }
+    {%- endif %}
+except environ.ImproperlyConfigured:
+    DATABASES = {
+        "default": {
+            "ENGINE": "django.db.backends.postgresql",
+            "NAME": env.str("POSTGRES_DB"),
+            "USER": env.str("POSTGRES_USER"),
+            "PASSWORD": env.str("POSTGRES_PASSWORD"),
+            "HOST": env.str("POSTGRES_HOST"),
+            "PORT": env.str("POSTGRES_PORT"),
+        }
+    }
```

Comment:
I'm not a big fan of this `except` 🤔 I can't quite really put a finger on why, though (sorry, that's unhelpful)

Reply:
I think it's because in case of problems, it because harder to figure out where the DB credentials are coming from. Do I have a `DATABASE_URL` set? Do I have anything missing from the required `POSTGRES_...` env? It's only one more place to check, but it can be _really_ confusing when your app won't conect to the DB

Reply:
Relaying on exceptions here is a lazy way of doing things I suppose. When a faulty url is given it would try to use the postgres vars, so I agree on your reluctance and it should be improved.